baseapp/baseapp.go

```diff
@@ -800,8 +796,7 @@ func (app *BaseApp) runMsgs(ctx sdk.Context, msgs []sdk.Msg, mode runTxMode) (*s
 			err          error
 		)
 
-		msgCtx := ctx.WithEventManager(sdk.NewEventManagerWithHistory(historicalEvents))
-		msgCtx = msgCtx.WithContext(context.WithValue(msgCtx.Context(), TxMsgIdxContextKey, i))
+		msgCtx := ctx.WithContext(context.WithValue(ctx.Context(), TxMsgIdxContextKey, i))
```

Comment:
Obviously out of scope for this PR, but should we make this more idiomatic in a followup?
```suggestion
		msgCtx := ctx.WithValue(TxMsgIdxContextKey, i)
```
The [implementation of `func (c Context) WithValue(key, value interface{})`](https://github.com/cosmos/cosmos-sdk/blob/v0.46.16/types/context.go#L262-L265) is **literally** `c.baseCtx = context.WithValue(c.baseCtx, key, value)`, which has the same effect as [`func (c Context) WithContext(ctx context.Context)`](https://github.com/cosmos/cosmos-sdk/blob/v0.46.16/types/context.go#L117)'s `c.baseCtx = ctx` with less external boilerplate.

Reply:
Why is this out of scope? Would it be changing the upstream source for improved readability/idiomatic-ness?

Reply:
Since we're touching the line anyhow, I got a scope waiver. Done.

Reply:
Richard and I considered it out-of-scope on the theory that a PR should stick to a single topic, for ease of review and future "blame"-viewing of the sources. Refactoring the tx-hash and msg-idx tracking seems like a different topic. But it's a judgment call, so since I was touching the line anyhow...